smarts/core/route.py

```diff
@@ -33,6 +43,10 @@ def edges(self) -> List[Edge]:
         """An (unordered) list of edges that this route covers"""
         raise NotImplementedError()
 
+    @property
+    def goalpoints(self) -> List[Any]:
```

Comment:
What is the rational of putting goal points as part of the route interface, why don't we just implement them as a sep. entity all together? Furthermore, I don't like that the goalpoints concept has worked its way all the way up to the top level route interface.

Reply:
There is no way to separate the points from the route interface since the checkpoints have to be on the route. If they are not on the route then you need to specify them separately from the route which forces changes on either the route or the checkpoints(which should be along the route.)